src/libponyc/codegen/genopt.cc

```diff
@@ -302,9 +315,9 @@ class HeapToStack : public FunctionPass
         }
 
         case Instruction::Load:
-          // This is a workaround for a problem with LLVM 4 & 5 on *nix when 
+          // This is a workaround for a problem with LLVM 4 & 5 on *nix when
           // hoisting loads (see #2303, #2061, #1592).
-          // TODO: figure out the real reason LLVM 4 and 5 produce bad code 
+          // TODO: figure out the real reason LLVM 4 and 5 produce bad code
```

Comment:
Have we figured out whether LLVM 6 also produces this bad code?

Reply:
Yes, there are still problems with LLVM 6 on Linux, but different from 4 & 5.  @Praetonus was going to look at it again in his copious free time.

Reply:
I turned on the workaround again for the time being, which is why the CI tests are passing.